cogs/displayinfo.py

```diff
@@ -0,0 +1,116 @@
+            else:
+                self.current_collection = db[mem_before.guild.name] # If the collection exists, get the existing collection for the guild
+                self.current_guild_name = mem_before.guild.name
+                self.timers = self.current_collection.find_one() # Load existing document from the MongoDB collection
+        
+        # Track activity updates for members
+        if not mem_before.activity and mem_after.activity: # activity starts/ongoing
+            if mem_after.name not in self.timers:
+                self.timers[mem_after.name] = {} # If the member is not already tracked, add them
+            if self.current_date not in self.timers[mem_after.name]:
+                self.timers[mem_after.name][self.current_date] = {} # If the current date is not present in the member's dictionary, add it
+            if mem_after.activity.name not in self.timers[mem_after.name][self.current_date]: # If the member's current activity/game/status is not present, add it with initial values
+                self.timers[mem_after.name][self.current_date][mem_after.activity.name] = {
+                    "totalTime" : 0, 
+                    "startTime" : 0,
+                }
+            self.timers[mem_after.name][self.current_date][mem_after.activity.name]["startTime"] = datetime.now() # Record the start time of the member's current activity
+            print("start:", self.timers)
+            asyncio.create_task(self.timed_update(mem_before, mem_after))
+        elif mem_before.activity and not mem_after.activity: # activity stopped
+            if mem_before.name in self.timers:
+                currentStartTime = self.timers[mem_before.name][self.current_date][mem_before.activity.name]["startTime"]
+                if currentStartTime:
+                    playTime = datetime.now() - currentStartTime # Calculate the time difference (playTime) between the start and end time of the activity
+                    self.timers[mem_before.name][self.current_date][mem_before.activity.name]["totalTime"] = self.timers[mem_before.name][self.current_date][mem_before.activity.name].get("totalTime", 0) + playTime.total_seconds() # Add the play time to the member's total play time for that activity for that day
+                print("end:", self.timers) #printing the dictionary to see the updates after an activity is ended
+                self.current_collection.replace_one({}, self.timers) # Update the MongoDB collection with the updated activity data(self.timers)
+
+    # This function is an asynchronous coroutine that runs indefinitely in the background when mem_after.activity is True.
+    # It is designed to update activity time information for a Discord member ('mem_after') whenever they start a new activity every 5 minutes.
+    async def timed_update(self, mem_before: discord.Member, mem_after: discord.Member):
```

Comment:
Might not need both parameters but just passing in the member is enough. After every 5 minutes, we should be able to be "safe" and just add the 5 minutes to their database because we should kill this coroutine once they stop their activity.

Reply:
This of course disregards the fail situations of the bot dying/going offline. We will need to research/discuss this more in the future. Please include this in description when creating the ticket regarding this.